client/src/partials/rubric/rubriques_payroll/rubriques_payroll.js

```diff
@@ -1,153 +1,157 @@
+    if(session.edit.is_advance && advance_exist){
+      messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_ADVANCE'));
+    } else if(session.edit.is_advance && !advance_exist  && !session.edit.is_discount){
+      messenger.danger($translate.instant('RUBRIC_PAYROLL.BE_DISCOUNT'));
+    } else {
+      var record = connect.clean(session.edit);
+      delete record.reference;
+      if (record.abbr) {
+        if (record.abbr.length <= 4) {
+          connect.put('rubric', [record], ['id'])
+          .then(function () {
+            vm.rubrics.put(record);
+            session.action = '';
+            session.edit = {};
+            messenger.success($translate.instant('RUBRIC_PAYROLL.UPDATE_SUCCES'));
+          })
+          .catch(error);
+        } else if (record.abbr.length > 4) {
+          messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_SUP4'));
         }
-      });
-
-      if(session.edit.is_advance && advance_exist){
-        messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_ADVANCE'));
-      } else if(session.edit.is_advance && !advance_exist  && !session.edit.is_discount){
-        messenger.danger($translate.instant('RUBRIC_PAYROLL.BE_DISCOUNT'));
       } else {
-        var record = connect.clean(session.edit);
-        delete record.reference;
-        if (record.abbr) {
-          if (record.abbr.length <= 4) {
-            connect.basicPost('rubric', [record], ['id'])
-            .then(function () {
-              messenger.success($translate.instant('RUBRIC_PAYROLL.UPDATE_SUCCES'));
-              $scope.rubrics.put(record);
-              session.action = '';
-              session.edit = {};
-            });
-          } else if (record.abbr.length > 4) {
-            messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_SUP4'));
-          }
-        } else {
-          messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_EMPTY'));
-        }
+        messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_EMPTY'));
       }
-    };
-
-    $scope.save.new = function () {
-      session.new.is_advance = (session.new.is_advance)?1:0;
-      session.new.is_percent = (session.new.is_percent)?1:0;
-      session.new.is_discount = (session.new.is_discount)?1:0;
+    }
+  }
 
-      var rubrics = $scope.rubrics.data,
+  function createSave () {
+    var rubrics = vm.rubrics.data,
         advance_exist = 0;
-      if(rubrics.length){
-        rubrics.forEach(function (rub) {
-          if(rub.is_advance){
-            advance_exist = 1;          
-          }
-        });
-      }  
-
-      if(session.new.is_advance && advance_exist){
-        messenger.danger($translate.instant('RUBRIC_PAYROLL.NOT_ADVANCE'));
-        session.new.is_advance = null;
-      } else if(session.new.is_advance && !advance_exist  && !session.new.is_discount){
-        messenger.danger($translate.instant('RUBRIC_PAYROLL.BE_DISCOUNT'));
-        session.new.is_advance = null;
+    if(rubrics.length){
```

Comment:
You don't need to check this `if` condition.  `rubrics.forEach()` will silently do nothing if `rubrics` is an empty array (`[]`). 

Also, you could use `var advance_exist = rubrics.some(fn);` for clarity.